Assignment1

```diff
@@ -0,0 +1,38 @@
+#include<iostream>
+using namespace std;
+int Array_Size;
+__global__ void Sum(int* d1_in, int* d2_in, int* d_out)
+{
+    int i = blockIdx.x * blockDim.x + threadIdx.x;
+        d_out[i] = d1_in[i] + d2_in[i];
```

Comment:
Here, you may add a condition,
if (i < d_Array_Size) // Take it as a parameter and also include proper declaration and allocation.
This is because in the last block may get extra threads and elements would be less than that. (To avoid overshooting the total size)